### ISSUE TITLE
NVDA does not announce all required details in JList (Java Access Bridge API)

### ISSUE THREAD

**Issue by anOolamr** [nvaccess/nvda#13907]
NVDA does not announce the details of a JLIst when no list item is selected.

### Steps to reproduce:

1. If not available, download the SwingSet2 Demonstration Program from: https://docs.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html#swingset2

2. Start NVDA.

3. Launch SwingSet2.jar with an up-to-date Java JRE.
-> The SwingSet2 application window appears.

4. Navigate with arrow right key to the "JList demo" radio button and press Spacebar key to select it.

5. Press Tab key twice to navigate with the keyboard focus to the JList control.
-> NVDA announces "list", which is the role of the JList control.
-> NVDA should announce the role and the number of list items and for example that no item is selected.

6. Press the command Ctrl + arrow down to set the keyboard focus on the first list item. 
-> NVDA announces "JoJoSystems, 1 of 30".

7. Press Ctrl + Tab to move the keyboard focus backwards out of the JList and then press Tab key to move the keyboard focus to the JList with the focused list item.
-> NVDA announces "list".
-> NVDA should announce "list, JoJoSystems, 1 of 30", as this item has the keyboard focus.

8. Press the command Ctrl + Spacebar to select the focused list item.
-> NVDA announces nothing.
-> NVDA should announce "JoJoSystems, selected, 1 of 30".

10. Press Ctrl + Tab to move the keyboard focus backwards out of the JList and then press Tab key to move the keyboard focus back to the JList with the selected list item.
-> NVDA announces "list, JoJoSystems, selected, 1 of 30".

11. Press the command Ctrl + Spacebar to deselect the list item.
-> NVDA announces nothing.
-> NVDA should announce "JoJoSystems, 1 of 30".

12. Press the command Ctrl + Spacebar to select the list item and then the command Shift + Arrow down to select another list item.
-> NVDA announces nothing.
-> NVDA should announce the two selected list items.

13. Press Ctrl + Tab to move the keyboard focus backwards out of the JList and then press Tab key to move the keyboard focus to the JList with the focused list item.
-> NVDA announces "list".
-> NVDA should announce the list role and the two selected list items.

### Actual behavior:
When navigating to a JList, NVDA does not announce list details except only one list item is selected.
NVDA does not announce the list item when selecting and deselecting it.
NVDA does not announce multiline-selection of list items.

### Expected behavior:
When navigating to a JLIst, NVDA announces also list details when no list item or more than one list item is selected.
NVDA announces the selection and deselection of list items.
NVDA announce multiline-selection of list items.

#### NVDA installed/portable/running from source:
installed

#### NVDA version:
Version: 2022.1

#### Windows version:
Version 10.0.19043 Build 19043

#### Name and version of other software in use when reproducing the issue:
java version "18.0.1" 2022-04-19
Java(TM) SE Runtime Environment (build 18.0.1+10-24)
Java HotSpot(TM) 64-Bit Server VM (build 18.0.1+10-24, mixed mode, sharing)

#### Other information about your system:
Java Access Bridge is enabled.

#### Does the issue still occur after restarting your computer?
Yws.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. Same behavior.

#### If NVDA add-ons are disabled, is your problem still occurring?
Not applicable.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not applicable.

**Comment by seanbudd** [comment]
>Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. Same behavior.

Which versions did you test? Have you tested 2022.2rc1?

**Comment by anOolamr** [comment]
@seanbudd I only used released version yet for testing (issue found in 2021.1/2021.2, retest in 2022.2).
Does the following build contain version 2022.2rc1?
alpha-25890,20774310	Mon Jul 18 02:25:45 2022
download from page:
https://www.nvaccess.org/files/nvda/snapshots/

If not, would you please provide a link to a build version of 2022.2rc1.
Thanks.

**Comment by Adriani90** [comment]
This is still an issue with NVDA 2024.4.1, and the same is valid for lists as well as for tree views. The number of selected items or the message that no item is selected should imo be claculated and exposed as message by the API. NVDA will probably not be able to calculate this itself without negative performance implications.
But maybe I am wrong and e.g. @dmitrii-drobotov maybe you have some thoughts on this as well?

It seems for jlists, navigating through selection with ctrl+arrow keys behave exactly like in Windows explorer.
1. selected items are reported without the atribute "selected" which is fine
2. Unselected items are reported as "not selected" which is fine as well.

When pressing ctrl+space bar, I guess the events and the messages "selected" or "not selected" need to be exposed properly by the API, same is valid for jtrees.
